src/pages/calendar/components/Requisites.tsx

```diff
@@ -0,0 +1,192 @@
+              <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                Earn a minimum grade of {req.grade} in each of:
+              </ListItem>
+            </UnorderedList>
+          )}
+          {reqs.map((r) => DisplayRequirement(r, indentationLevel + 1))}{' '}
+          {/* Ensure to increase the indentation level for nested elements */}
+        </Box>
+      );
+    } else {
+      return (
+        <Box>
+          {req.quantity &&
+            !req.grade &&
+            indentationLevel === 1 && ( // Displays this only when its not a grade requirement, and is at the lowest indentation level
+              <UnorderedList>
+                <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                  Complete {req.quantity} of the following:{' '}
+                </ListItem>
+              </UnorderedList>
+            )}
+          {req.quantity &&
+            !req.grade &&
+            indentationLevel !== 1 && ( // Displays this only when its not a grade requirement
+              <UnorderedList>
+                <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>Complete {req.quantity} of: </ListItem>
+              </UnorderedList>
+            )}
+          {req.grade && (
+            <UnorderedList>
+              <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                Earn a minimum grade of {req.grade} in each of the following:
+              </ListItem>
+            </UnorderedList>
+          )}
+          {reqs.map((r) => DisplayRequirement(r, indentationLevel + 1))}{' '}
+          {/* Ensure to increase the indentation level for nested elements */}
+        </Box>
+      );
+    }
+  } else if ('reqListItems' in req) {
+    const nestedReqs = nestedReq(req);
+    return <Box>{nestedReqs}</Box>; // Displays the list of requisites
+  } else if ('code' in req) {
+    // Get the current term and requisite details
+    const subject = req.subject;
+    const code = req.code;
+
+    // Get course details from the backend
+    return (
+      <GetCourseDetails term={currTerm} subject={req.subject} code={req.code}>
+        {(courseDetails, isLoading) => {
+          // If its loading then render a skeleton
+          if (isLoading.loading) {
+            return (
+              <UnorderedList>
+                <ListItem
+                  title={`${subject} ${code}`}
+                  style={{ marginLeft: `${indentationLevel * 40}px`, marginRight: `100px` }}
+                >
+                  <Skeleton>
+                    <Text>
+                      Loading course details of {req.subject} {req.code}
+                    </Text>
+                  </Skeleton>
+                </ListItem>
+              </UnorderedList>
+            );
+          } else if (courseDetails) {
+            // Extract course details and format credits
+            const pid = courseDetails.pid;
+            const credits = courseDetails.credits.credits;
```

Comment:
```suggestion
            const pid = courseDetails.pid;
            const credits = courseDetails.credits.credits;
            const {pid, credits: {credits}} = courseDetails;
```
might have to check that one, can't remember syntax for multi-destructuring